projects/dex-ui/src/components/Liquidity/allowance.ts

```diff
@@ -1,35 +1,19 @@
 import { ERC20Token } from "@beanstalk/sdk-core";
 import { BigNumber } from "ethers";
+import { Log } from "src/utils/logger";
 
 const hasMinimumAllowance = async (walletAddress: string, spender: string, token: ERC20Token, mininumAllowance: BigNumber) => {
-  const existingAllowance = await token.getContract().allowance(walletAddress, spender);
-  
-  if (!existingAllowance) {
-    throw new Error("Unable to retrieve allowance");
-  }
-  
-  return existingAllowance.gte(mininumAllowance);
+  const existingAllowance = await token.getAllowance(walletAddress, spender);
+  return existingAllowance.toBigNumber().gte(mininumAllowance);
```

Comment:
Not sure if you need to go toBigNumber() here. I mean, you do because mininumAllowance is a BigNumber, but the only reason it's that is because you converted a TokenValue to bigNumber when you passed it in. TLDR, If you're working with TokenValues, you should never need to convert to bignumber unless your sending that value to a lower level block chain thing, like ethers or something